src/gentropy/dataset/summary_statistics.py

```diff
@@ -114,3 +114,26 @@ def exclude_region(self: SummaryStatistics, region: str) -> SummaryStatistics:
             ),
             _schema=SummaryStatistics.get_schema(),
         )
+
+    def sanity_filter(self: SummaryStatistics) -> SummaryStatistics:
+        """The function filters the summary statistics by sanity filters.
+
+        The function filters the summary statistics by the following filters:
+            - The p-value should not be eqaul 1.
+            - The beta and se should not be equal 0.
+            - The p-value, beta and se should not be NaN.
+
+        Returns:
+            SummaryStatistics: The filtered summary statistics.
+        """
+        gwas_df = self._df
+        gwas_df = gwas_df.dropna(
+            subset=["beta", "standardError", "pValueMantissa", "pValueExponent"]
+        )
+
+        gwas_df = gwas_df.filter((f.col("beta") != 0) & (f.col("standardError") != 0))
+        gwas_df = gwas_df.filter(
+            f.col("pValueMantissa") * 10 ** f.col("pValueExponent") != 1
+        )
+        self._df = gwas_df
+        return self
```

Comment:
This is very dangerous. What this statement does is that it returns the updated self. How it works:
```python
unfiltered_summary_stats.df.count() # -> 1000
filtered_summary_stats = unfiltered_summary_stats.sanity_filter()

filtered_summary_stats.df.count() # -> 500
unfiltered_summary_stats.df.count() # -> 500
```
Because there will be no `unfiltered_summary_stats` anymore, just the filtered. This behaviour can lead to problems that are very hard to track. (been there, done that.)

Instead we have to do this:
```python
        return SummaryStatistics(
            _df=(
                self.df.dropna(
                    subset=["beta", "standardError", "pValueMantissa", "pValueExponent"]
                )
                .filter(
                    (f.col("pValueMantissa") * 10 ** f.col("pValueExponent") != 1)
                    & (f.col("beta") != 0)
                    & (f.col("standardError") != 0)
                )
            ),
            _schema=SummaryStatistics.get_schema(),
        )
```
The above statement returns a completely new dataset so the old, the unfiltered still remains.

Reply:
Done.